jwst/outlier_detection/outlier_detection.py

```diff
@@ -206,38 +205,29 @@ def do_detection(self):
                     good_bits=pars['good_bits'])
 
         # Initialize intermediate products used in the outlier detection
-        median_model = datamodels.ImageModel(drizzled_models[0].data.shape)
-        median_model.update(drizzled_models[0])
-        median_model.meta.wcs = drizzled_models[0].meta.wcs
+        dm0 = datamodel_open(drizzled_models[0])
+        median_model = datamodels.ImageModel(dm0.data.shape)
+        # median_model = datamodels.ImageModel(drizzled_models[0].data.shape)
+        median_model.update(dm0)
+        median_model.meta.wcs = dm0.meta.wcs
+        dm0.close()
+        del dm0
 
         # Perform median combination on set of drizzled mosaics
         median_model.data = self.create_median(drizzled_models)
-
-        if save_intermediate_results:
-            median_output_path = self.make_output_path(
-                basepath=self.input_models[0].meta.filename,
-                suffix='median'
-            )
-            log.info("Writing out MEDIAN image to: {}".format(
-                median_output_path
-            ))
-            median_model.save(median_output_path)
-
+        median_model_output_path = self.make_output_path(
+            basepath=median_model.meta.filename,
+            suffix='median')
+        median_model.save(median_model_output_path)
```

Comment:
So the median image is just saved all the time now, rather than being controlled by a step param? I assume that's necessary in order to keep memory usage down?

Reply:
That is correct.